elyra/pipeline/processor.py

```diff
@@ -605,3 +608,31 @@ def _get_volume_mounts(self, operation: Operation) -> Optional[Dict[str, str]]:
                 formatted_mount_path = f"/{mount_path.strip('/')}"
                 volume_mounts_valid[formatted_mount_path] = pvc_name
         return volume_mounts_valid
+
+    def _get_kubernetes_secrets(self, operation: Operation) -> Optional[List[Tuple]]:
+        """
+        Loops through an Operation's kubernetes secrets to strip whitespace
+        and re-format as a tuple.
+
+        :param operation: the operation to check for secrets
+        :return: tuple of env var name, secret name, and secret key
+        """
+        valid_secrets = []
+        secrets = operation.component_params.get(KUBERNETES_SECRETS)
+        if secrets and isinstance(secrets, KeyValueList):
+            for env_var_name, secret in secrets.to_dict().items():
+                secret_tuple = secret.split(":", 1)
+                if len(secret_tuple) != 2:
+                    self.log.warning(f"Ignoring invalid secret for '{env_var_name}': missing secret name and/or key.")
+                    continue
+                secret_name, secret_key = secret_tuple[0].strip(), secret_tuple[1].strip()
+                if not secret_name:
```

Comment:
We also need to make sure that the user-provided secret name meets the Kubernetes naming convention. If it doesn't, we have to ignore it because it would cause a failure during pipeline execution. See https://github.com/elyra-ai/elyra/blob/master/elyra/util/kubernetes.py for related code

Reply:
I was going to ask you about this - I'll add in some calls to this utility function!